File: python3/781.py
class Solution:
    def numRabbits(self, r: List[int]) -> int:
        c = 0
        f = [0] * 1000
        for i in r:
            if f[i] > 0:
                f[i] -= 1
            else:
                c += i + 1
                f[i] = i
        return c

File: java/781.java
class Solution {
    public int numRabbits(int[] r) {
        int c = 0, n = r.length;
        int f[] = new int[1000];
        for (int i = 0; i < n; i++)
            if (f[r[i]] > 0)
                f[r[i]]--;
            else {
                c += r[i] + 1;
                f[r[i]] = r[i];
            }
        return c;
    }
}
